File: app/src/main/java/com/openmobl/pttDriver/bt/SerialListener.java
package com.openmobl.pttDriver.bt;

import java.util.UUID;

public interface SerialListener {
    void onSerialConnect();
    void onSerialConnect(UUID service, UUID characteristic);
    void onSerialConnectError(Exception e);
    void onSerialDisconnect();
    void onSerialRead(byte[] data, UUID service, UUID characteristic);
    void onSerialIoError(Exception e);
    void onBatteryEvent(byte level);
}

File: app/src/main/java/com/openmobl/pttDriver/bt/SerialListener.kt
package com.openmobl.pttDriver.bt

import java.util.*

interface SerialListener {
    fun onSerialConnect()
    fun onSerialConnect(service: UUID?, characteristic: UUID?)
    fun onSerialConnectError(e: Exception)
    fun onSerialDisconnect()
    fun onSerialRead(data: ByteArray?, service: UUID?, characteristic: UUID?)
    fun onSerialIoError(e: Exception)
    fun onBatteryEvent(level: Byte)
}